main.tf

```diff
@@ -1,65 +1,65 @@
-provider "aws" {
+  provider "aws" {
   region = "us-east-1"
 
 }
 
-resource "aws_instance" "app-server" {
-  instance_type = var.ec2-instance
+resource "aws_instance" "app-server" {          //zz problem, fix this by enabling kk for a resource with a module path: root             || for Non-attribute problems
+  instance_type = var.ec2-instance              //Change this to xx to save yy dollars annually for a resource with a module path: root      ||note: this var here in root is redundant & implies default value, otherwise vars are not going to get linted, only values
```

Comment:
Can't understand what this means "note: this var here in root is redundant & implies default value, otherwise vars are not going to get linted, only values"

Can you state it in terms of. If the developer had not put a var here and instead had doxn xyz then Abc would have happened.

Even better, just create that case and put a comment on it.

Reply:
yes, addressed from user's pov